src/types/query-types.ts

```diff
@@ -10,7 +10,9 @@ export enum SortDirection {
   Ascending = 1
 }
 
-export type EqualFilter = string | number | boolean;
```

Comment:
I see the intent of wanting to be more specific with indexed value types. But `FilterIndex` seems to vague of an abstraction name for this, code just doesn't read well with it.

A alternative that shows the intent could be to use this type instead:

`type Indexes = { [key: string]: string | number | boolean };`

Or more generically:

`type KeyValues = { [key: string]: string | number | boolean };`

Or have both being the same type like what you've done for `EqualFilter` and `FilterIndex` and use the "right" one in their context.

Reply:
💯  This is much better, fixed.